source/Patches/CrewmateRoles/ProphetMod/Revelation.cs

```diff
@@ -1,44 +1,35 @@
-using System;
+using System.Linq;
 using HarmonyLib;
 using TownOfUs.Roles;
 
 namespace TownOfUs.CrewmateRoles.ProphetMod
 {
-    [HarmonyPatch(typeof(PlayerControl), nameof(PlayerControl.FixedUpdate))]
+    [HarmonyPatch(typeof(PlayerControl), nameof(PlayerControl.CompleteTask))]
     public class Revelation
     {
         public static void Postfix(PlayerControl __instance)
         {
-            if (
-                PlayerControl.AllPlayerControls.Count <= 1
-                || PlayerControl.LocalPlayer == null
-                || PlayerControl.LocalPlayer.Data == null
-                || MeetingHud.Instance
-                || !PlayerControl.LocalPlayer.CanMove
-                || !PlayerControl.LocalPlayer.Is(RoleEnum.Prophet)
-                || PlayerControl.LocalPlayer.Data.IsDead
-                )
+            if (!PlayerControl.LocalPlayer.Data.IsDead
+                && PlayerControl.LocalPlayer.Is(RoleEnum.Prophet)
+                && ShouldReveal(__instance))
             {
-                return;
+                Role.GetRole<Prophet>(PlayerControl.LocalPlayer).Revelation();
             }
-
-            if (!IsTimeForRevelation())
-            {
-                return;
-            }
-
-            Role.GetRole<Prophet>(PlayerControl.LocalPlayer).Revelation();
         }
 
-        private static bool IsTimeForRevelation()
+        private static bool ShouldReveal(PlayerControl __instance)
         {
-            var now = DateTime.UtcNow;
+            var taskInfos = __instance.Data.Tasks.ToArray();
+            var allTasksCount = taskInfos.Count;
+            var maxRevealsCount = 
+                CustomGameOptions.ProphetTotalReveals < 1 ? 1 : CustomGameOptions.ProphetTotalReveals;
+            var currentRevealsCount = Role.GetRole<Prophet>(PlayerControl.LocalPlayer).Revealed.Count;
```

Comment:
This is using `PlayerControl.LocalPlayer` but the method takes in a `PlayerControl __instance` object. Not sure which is intended.

Reply:
I modelled it after `CompleteTask.cs` from the Snitch, where it seems like both patterns are used. I think they both work, but not sure what the implications are. I tried a bunch of different multi-client setups to test and didn't run into any problems as coded.

Reply:
Also just removed the `__instance` and it still works. It seems like both `PlayerControl.LocalPlayer` and `__instance` are really the same thing.